From 1359a849055a66eddffe7753485e0f29fb85ef31 Mon Sep 17 00:00:00 2001
From: Eric Blake <eblake@redhat.com>
Date: Fri, 4 Jun 2010 06:07:38 -0600
Subject: [PATCH 1/3] Fix build on newer glibc.
Bug-Debian: http://bugs.debian.org/584777
X-Debian-version: 1.4.14-3

POSIX allows, but does not require, that <fcntl.h> implicitly
includes <sys/stat.h>.  Until recent glibc, most systems did
this implicit inclusion, and m4 was mistakenly relying on it.

* src/m4.h (includes): Add <sys/stat.h>.
Reported by Todd Rinaldo.

Signed-off-by: Eric Blake <eblake@redhat.com>
---
 src/m4.h  |    1 +
 3 files changed, 10 insertions(+), 0 deletions(-)

--- a/src/m4.h
+++ b/src/m4.h
@@ -32,6 +32,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <string.h>
+#include <sys/stat.h>
 #include <sys/types.h>
 
 #include "binary-io.h"
